### PR TITLE
Replace BufferedAtomics with UnorderedAtomics

### PR DESCRIPTION
**Pull request by ronawho**
Replace the `BufferedAtomics` module (`amoBuff()` / `flushAtomicBuff()`) with
an `UnorderedAtomics` module (`unorderedAmo()` / `unorderedAtomicFence()`). The
buffered API exposed too much about the implementation and didn't follow
our normal naming scheme.

The UnorderedAtomics module is effectively a code clone of BufferedAtomics with
documentation updated and new function names. Beyond that, this changes all
existing uses of buffered atomics to unordered atomics and renames tests,
graphkeys, etc.

Note that the BufferedAtomics is being deprecated, but not removed (we
documented the API as being unstable, but I figured deprecating can't hurt.)

Longer term I expect things like `Atomic.unorderedAdd(val)` to become
`Atomic.add(val, order=unordered)`, but I'm not ready to commit to a
language-level API yet, and this is at least a stepping stone.

### CODE REVIEW

**Comment by ronawho** [comment]
There's a decent amount of code here, but it's almost entirely just renaming tests/functions/config-vars. The module code change is a code clone with documentation and function names updated.

TODO (reminder for myself) -- rename the nightly .dat files

Rendered docs available at: http://web/~eronagha/unordered-atomics-docs/modules/packages/UnorderedAtomics.html

File: modules/packages/UnorderedAtomics.chpl [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+     // no fence required, fenced at task termination
+
+     const itersSum = iters*(iters+1)/2, // sum from 1..iters
+           numTasks = numLocales * numTasksPerLocale;
+     assert(a.read() == numTasks * itersSum);
+
+   It's important to be aware that unordered atomic operations are not
+   consistent with regular atomic operations and updates may not be visible
+   until the task that issued them terminates or they are explicitly fenced
+   with :proc:`unorderedAtomicFence()`.
```
Comment (mppf) [inline modules/packages/UnorderedAtomics.chpl]: Should it mention the fence at the end of a forall ? And mention the fence at task termination in the prose?

Comment (ronawho) [reply modules/packages/UnorderedAtomics.chpl]: Task termination fence is already mentioned, but I will add a note about forall

**Comment by ronawho** [comment]
> I thought we'd end up with `myAtomic.add(order=unordered)` ? Did we decide against that? If not, maybe the API will change yet again?

Yeah, eventually I think we'll end up with something like that, but I'm not ready to commit to something that's effectively part of the language yet, since that makes it much harder to make changes. 

I updated the description to say "Longer term I expect things like `Atomic.unorderedAdd(val)` to become `Atomic.add(val, order=unordered)`, but I'm not ready to commit to a language-level API yet, and this is at least a stepping stone.`

File: modules/packages/UnorderedAtomics.chpl (hunk 1 above)
Comment (ronawho) [reply modules/packages/UnorderedAtomics.chpl]: Added note about fence at end of forall (and regenerated docs)

**Pull request merged**